src/software/sensor_fusion/sensor_fusion.h

```diff
@@ -40,17 +40,24 @@ class SensorFusion : public Subject<World>, public ThreadedObserver<SensorMsg>
     void onValueReceived(SensorMsg sensor_msg) override;
 
     /**
-     * Updates world based on a new data
+     * Updates components of world based on a new data and sends World to observers if
+     * complete
      *
      * @param new data
      */
     void updateWorld(const SensorMsg &sensor_msg);
-    void updateWorld(const SSL_WrapperPacket &packet);
-    void updateWorld(const Referee &packet);
-    void updateWorld(
+
+    /**
+     * Updates relevant components of world based on a new data
+     *
+     * @param new data
+     */
+    void updateWorldComponents(const SSL_WrapperPacket &packet);
+    void updateWorldComponents(const Referee &packet);
+    void updateWorldComponents(
         const google::protobuf::RepeatedPtrField<TbotsRobotMsg> &tbots_robot_msgs);
-    void updateWorld(const SSL_GeometryData &geometry_packet);
-    void updateWorld(const SSL_DetectionFrame &ssl_detection_frame);
+    void updateWorldComponents(const SSL_GeometryData &geometry_packet);
+    void updateWorldComponents(const SSL_DetectionFrame &ssl_detection_frame);
```

Comment:
Hmm, I think I prefer just updateWorld, I don't think `Components` really adds a ton of value here?

Reply:
removed